packages/cloud/src/queries/tenants.ts

```diff
@@ -109,6 +122,33 @@ export const createTenantsQueries = (client: Queryable<PostgreSql>) => {
     return rows;
   };
 
+  const deleteAdminTenantM2MApplicationByPrincipal = async (tenantId: string) => {
+    await client.query(sql`
+      delete from applications where custom_client_metadata->>'tenantId' = ${tenantId} and tenant_id = ${adminTenantId} and "type" = ${ApplicationType.MachineToMachine}
+    `);
+  };
+
+  const deleteAdminTenantResourcesByIndicator = async (indicator: string) => {
```

Comment:
nit: this could also use a similar pattern.  deleteClientTenantManagementApiResourceByTenantId ?

Reply:
Updated.